fairing/ml_tasks/tasks.py

```diff
@@ -64,8 +65,18 @@ def create(self):
         self.url = self._deployer.deploy(self.pod_spec)
         logger.warning("Prediction endpoint: {}".format(self.url))
 
-    def predict(self, data):
-        r = requests.post(self.url, data=data)
+    def predict_nparray(self, data, names=["t:0", "t:1"]):
```

Comment:
Seldon expects the request body to be [SeldonMessage](https://github.com/SeldonIO/seldon-core/blob/36e96c162bfaa5c1fb5a3286d8c02dba8e393071/util/loadtester/scripts/proto/prediction.proto) proto. The names field represents the column/feature names.  Therefore we shouldn't default to any value here other than empty array or none. names here is passed to predict function as feature_names. However there is no check on len(names)==data.shape[1] in their code.
Ref for future reference to improve client impl:
https://github.com/SeldonIO/seldon-core/blob/7ad5eee98c2724c8ca27e9877304ad8c27a6ed84/python/seldon_core/seldon_methods.py#L38 
https://github.com/SeldonIO/seldon-core/blob/7ad5eee98c2724c8ca27e9877304ad8c27a6ed84/python/seldon_core/utils.py#L363
https://github.com/SeldonIO/seldon-core/blob/bb9ce7fd7aece435100ceb720bc9a5f421de786a/python/seldon_core/user_model.py#L133

Reply:
done